### ISSUE TITLE
VirtualTable with Lazy Loading and rootComponent height override resets scroll position

### ISSUE THREAD

**Issue by jkernagis2** [DevExpress/devextreme-reactive#2648]
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Current Behaviour

When you use the Grid component's rootComponent attribute to provide a root component override, and set the height style (hard coded height or to 100% as in the virtual scrolling, "Fill the Container" example: https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/virtual-scrolling/), the scroll position resets when you scroll down far enough to cause additional rows to load.

### Expected Behaviour

The scroll position should not reset while scrolling down.

### Steps to Reproduce

[codesandbox](https://codesandbox.io/s/devextreme-react-grid-for-material-ui-guyfx)).

Using base examples for virtual table with lazy loading, add a rootComponent override to the Grid to specify height style.  I used a hardcoded 300px in the codepen, but would like to have this set to 100% in our application.

Scroll down in the virtual table, observe scroll position being reset to the top when additional data is attempted to be loaded.

ADDITIONAL INFO:
I've found that if I manually (using chrome devtools) add the style="height: 300px" or style="height: 100%" to the root div (div class="RootBase-root-###"), the height adjusts as expected, and the scroll position reset bug  *does not occur*.

Unless I'm missing something, I don't see anything different in the DOM between manually setting that style and setting it with the rootComponent override as in the codepen.

### Environment

See codepen.

**Comment by AryamnovEugeniy** [comment]
Hello,

I noticed that in your example you declare your root component inside the `render` method. Doing so, you create it from scratch every time the Grid rerenders. This may lead to incorrect behaviour and performance losses. We advise not to declare neither functions nor variables inside the `render` method (here is our [article](https://devexpress.github.io/devextreme-reactive/react/common/docs/guides/performance-optimization/) on how to optimize performance).
And here I made an [example](https://codesandbox.io/s/devextreme-react-grid-for-material-ui-jljjm) where the issue you have described is not reproduced. In my example I create the `Root` component outside the `render` method.

If you have any other questions, feel free to ask them.
